src/Database/LSMTree/Monoidal.hs

```diff
@@ -97,126 +131,178 @@ closeTableSession = undefined
 -- an LSM table. The multiple-handles feature allows for there to may be many
 -- such instances in use at once.
 type TableHandle :: (Type -> Type) -> Type -> Type -> Type
-newtype TableHandle m k v = TableHandle {
-    tsSession :: TableSession m k v
+data TableHandle m k v = TableHandle {
+    thSession :: !(Session m)
+  , thConfig  :: !TableConfig
   }
 
--- | Create a new table referenced by a table handle.
+-- | Table configuration parameters, like tuning parameters.
 --
--- NOTE: close table handles using 'closeTableHandle' as soon as they are
--- unused.
-newTable ::
-     IOLike m
-  => TableSession m k v
-  -> m (TableHandle m k v)
-newTable = undefined
+-- Some config options are fixed (for now):
+-- * Merge policy: Tiering
+-- * Size ratio: 4
+data TableConfig = TableConfig {
+    -- | Total number of bytes that the write buffer can use.
+    tcMaxBufferMemory      :: Word64
+    -- | Total number of bytes that bloom filters can use collectively.
+  , tcMaxBloomFilterMemory :: Word64
+    -- | Bit precision for the compact index
+    --
+    -- TODO: fill in with a realistic, non-unit type.
+  , tcBitPrecision         :: ()
+  }
+
+-- | Configs should be comparable, because only tables with the same config
+-- options are __compatible__.
+--
+-- For more information about compatibility, see 'Session'.
+deriving instance Eq TableConfig
 
--- | This is a temporary API for while snapshots aren't implemented yet.
--- Eventually, we should use 'loadSnapshot' instead.
+-- | Create a new table referenced by a table handle.
 --
 -- NOTE: close table handles using 'closeTableHandle' as soon as they are
 -- unused.
-openTable ::
+new ::
      IOLike m
-  => TableSession m k v
-  -> m (Either
-          VerificationFailure
-          (TableHandle m k v)
-       )
-openTable = undefined
+  => Session m
+  -> TableConfig
+  -> m (TableHandle m k v)
+new = undefined
 
 -- | Close a table handle.
 --
 -- Any on-disk files and in-memory data that are no longer referenced after
 -- closing the table handle are lost forever. Use 'Snapshot's to ensure data is
 -- not lost.
-closeTable ::
+--
+-- TOREMOVE: before snapshots are implemented, 'open' should still work after
+-- closing a table handle.
```

Comment:
I don't think I understand this comment.

Reply:
Removed it